### PR TITLE
Add QTermWidget::saveHistory and QTermWidget::sendCommand.

### PR DESCRIPTION
**Pull request by Andersbakken**
This enables an application to implement the functionality of easily
opening the entire history as a file in your favorite editor.

### CODE REVIEW

File: lib/qtermwidget.cpp [hunk 1]
```diff
@@ -749,3 +749,16 @@ int QTermWidget::getMargin() const
...
+{
+    QTextStream stream(device);
+    PlainTextDecoder decoder;
+    decoder.begin(&stream);
+    m_impl->m_session->emulation()->writeToStream(&decoder, 0, m_impl->m_session->emulation()->lineCount());
+}
+
+void QTermWidget::sendCommand(const QString &command)
+{
+    m_impl->m_session->emulation()->sendText(command);
```
Comment (yan12125) [inline lib/qtermwidget.cpp]: Well, this is basically the same as `QTermWidget::sendText()`.

Comment (Andersbakken) [reply lib/qtermwidget.cpp]: It is indeed. I've just removed it now. Thanks

**Comment by agaida** [comment]
@yan12125 - would you be so kind and check again?

**Comment by Andersbakken** [comment]
I'm a little confused why this still is marked as change requested. I thought that was for the removal of sendCommand

**Comment by agaida** [comment]
@Andersbakken - i guess @yan12125 has to confirm that his request is solved

**Comment by yan12125** [comment]
Thanks for the update and sorry for being late. My original plan was to merge this PR and the one at qterminal at the same time, but there are some issues in the qterminal PR. I will post my observations there as soon as possible.

**Pull request merged**